rosidl_generator_py/resource/_msg.py.em

```diff
@@ -182,45 +186,36 @@ class @(message.structure.type.name)(metaclass=Metaclass_@(message.structure.typ
 @[end for]@
     ]
 
-    _fields_and_field_types = {
+    SLOT_TYPES = (
 @[for member in message.structure.members]@
 @{
 type_ = member.type
 if isinstance(type_, NestedType):
     type_ = type_.basetype
 }@
-        '@(member.name)': '@
-@# the prefix for nested types
-@[  if isinstance(member.type, Sequence)]@
-sequence<@
+        @
+@[  if isinstance(member.type, Sequence) or isinstance(member.type, Array)]@
+@(member.type.__class__.__module__).@(member.type.__class__.__name__)(@
 @[  end if]@
 @# the typename of the non-nested type or the nested basetype
+@(type_.__class__.__module__).@(type_.__class__.__name__)(@
 @[  if isinstance(type_, BasicType)]@
-@(type_.type)@
+'@(type_.type)')@
 @[  elif isinstance(type_, BaseString)]@
-@
-@[    if isinstance(type_, WString)]@
-w@
-@[    end if]@
-string@
-@[    if type_.maximum_size is not None]@
-<@(type_.maximum_size)>@
-@[    end if]@
+'@(str(type_.maximum_size))')@
 @[  elif isinstance(type_, NamespacedType)]@
-@('/'.join([type_.namespaces[0], type_.name]))@
+['@("', '".join(type_.namespaces))'], '@(type_.name)')@
```

Comment:
This is wrong if `namespaces` is empty.

Reply:
In which scenario would someone use a NamespacedType without namespace?

---

Is the expected syntax the following ?
```
NamespacedType([], 'myname')
```
instead of 
```
NamespacedType([''], 'myname')
```

---

I can fix it in here, wouldn't the other places in this file using this logic need to be updated as well ?
e.g. https://github.com/ros2/rosidl_python/blob/4fe21665960c3fefad5d46d3a6e559b39598018f/rosidl_generator_py/resource/_msg.py.em#L247

Reply:
> In which scenario would someone use a NamespacedType without namespace?

There isn't a specific case where it is used atm but the code could be defensive and not rely on that assumption.

> `NamespacedType([], 'myname')`

Correct.

> wouldn't the other places in this file using this logic need to be updated as well ?

Probably, doesn't have to happen in this PR though. It's more a future proofing thing.